core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -2513,7 +2513,8 @@ private void sqlShow(SqlExecutionContext executionContext) throws SqlException {
             if (factory != null) {
                 tok = SqlUtil.fetchNext(lexer);
                 if (tok == null || Chars.equals(tok, ';')) {
-                    compiledQuery.of(factory);
+                    short type = factory instanceof NotCachedRecordCursorFactory ? CompiledQuery.PSEUDO_SELECT : CompiledQuery.SELECT;
```

Comment:
sorry I noticed late, there is a pattern used by `RecordCursorFactory` that instead of marker interface there are flag methods with default implementation on factory intself.

The reason it is the case is that Java has performance bugs around `instance of` and we are trying to avoid using it

Reply:
> The reason it is the case is that Java has performance bugs around `instance of` and we are trying to avoid using it

In fact, that's not the only reason. As @jerrinot pointed, the marker interface approach will break on nested queries, JOINs, UNIONs and other types of SELECTs where the SHOW statement acts as a sub-query. On the other hand, the flag method won't have this problem as long as we override it correctly here and there.

Reply:
fixing